crates/astria-sequencer/src/ibc/ics20_withdrawal.rs

```diff
@@ -59,30 +57,36 @@ fn withdrawal_to_unchecked_ibc_packet(
     )
 }
 
-async fn ics20_withdrawal_check_stateful_bridge_account<S: StateRead>(
+// bridge address checks:
+// - if the sender of this transaction is not a bridge account, and the tx `bridge_address` field is
+//   None, don't need to do any bridge related checks as it's a normal user withdrawal.
+// - if the sender of this transaction is a bridge account, and the tx `bridge_address` field is
+//   None, check that the withdrawer address is the same as the transaction sender.
+// - if the tx `bridge_address` field is Some, check that the `bridge_address` is a valid bridge,
+//   and check that the withdrawer address is the same as the transaction sender.
```

Comment:
move this to above `check_and_execute`?

Reply:
This was shuffling around code. I forgot to remove it.